Source/WebGPU/WebGPU/BindGroupLayout.h

```diff
@@ -55,7 +56,12 @@ class BindGroupLayout : public WGPUBindGroupLayoutImpl, public RefCounted<BindGr
         BindingLayout bindingLayout;
     };
 
-    static Ref<BindGroupLayout> create(HashMap<uint32_t, WGPUShaderStageFlags>&& stageMapTable, id<MTLArgumentEncoder> vertexArgumentEncoder, id<MTLArgumentEncoder> fragmentArgumentEncoder, id<MTLArgumentEncoder> computeArgumentEncoder, Vector<Entry>&& entries)
+    struct BufferInfo {
+        using IndexType = uint32_t;
```

Comment:
We can certainly use 64-bit types (NSUInteger), but I think WebGPU only supports 32-bit types. Happy to change this to NSUInteger if preferred

Reply:
I think I prefer it from a readability standpoint - it makes it clear that this is an index into a platform data structure, rather than a WebGPU data structure.